lib/python/treadmill_aws/hostmanager.py

```diff
@@ -1,35 +1,51 @@
 """ Module defining interface to create/delete/list IPA-joined hosts on AWS.
 """
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
 import time
 
 from treadmill_aws import ec2client
 
 
-def render_manifest(hostname, otp):
+def render_manifest(key_value_pairs, url_list=None):
     """ Stub function to supply instance user_data during testing. """
-    template = '''#cloud-config
-#
-# install ipa-client
+    combined_userdata = MIMEMultipart()
+
+    # Generate k:v formatted string from input variables
+    kv_template = ['{key}: {val}'.format(key=key, val=val)
```

Comment:
why not yaml.dump?

Reply:
No real objection- didn't want to bring in the whole module for one operation.